0015-estimator-interface.md

```diff
@@ -0,0 +1,457 @@
+
+We now describe the strategy in detail, beginning with a restatement of the current interface, where `Estimator.run` has `circuits` as the only positional argument and accepts `observables` and `parameter_values` as keyword arguments (in addition to the var keyword `**run_options` arguments which isn't affected):
+
+```python
+class Estimator(BasePrimitive):
+
+    # current signature
+    def run(
+            self,
+            circuits: Sequence[QuantumCircuit] | QuantumCircuit,
+            observables: Sequence[BaseOperator | PauliSumOp | str] | BaseOperator | PauliSumOp | str,
+            parameter_values: Sequence[Sequence[float]] | Sequence[float] | float | None = None,
+            **run_options,
+        ):
+        ...
+```
+
+We propose the migration to using tasks in two phases.
+
+### Deprecation Phase 1
+
+In the first phase:
+* Introduce `tasks: Sequence[EstimatorTask] | EstimatorTask` as the only positional argument, and make `circuits` a keyword argument.
+* Coerce the arguments to account for the fact that the only existing positional argument is `circuit: Sequence[QuantumCircuit] | QuantumCircuit`.
+* Check that the user does not attempt to mix the old/new APIs.
+* If necessary, coerce the old-API arguments (now all keyword arguments) into `EstimatorTask`s, **raise deprecation warning**.
+* Always eventually run with `Estimator._run(tasks: Sequence[EstimatorTask], **run_options)`.
+
+Here is a mock implementation, with missing type annotations as above:
+```python
+class Estimator(BasePrimitive):
+
+    def run(
+        self, 
+        tasks: Sequence[EstimatorTask] | EstimatorTask, 
+        circuits = None,
+        observables = None, 
+        parameter_values = None, 
+        **run_options
+    ):
+        # Coerce into standard form to later allow for `tasks` to be the only non-kwarg.
+        if isinstance(tasks, QuantumCircuit) or all(isinstance(task, QuantumCircuit) for task in tasks):
+            if circuits is not None:
+                raise ValueError("Cannot mix old and new APIs")
+            circuits = tasks
+            tasks = None
+
+        # Do not mix new/old API
+        if tasks and (circuits or observables or parameter_values):
+            raise ValueError("Cannot mix old and new APIs")
+
+        # Deprecated path
+        if tasks is None:
+            # Verify values in old API
+            if not (circuits and observables):
+                raise ValueError(f"Need both circuits and observables for old API, got circuits={circuits}, observables={observables}")
+            
+            # Coerce into old form
+            if parameter_values is None:
+                parameter_values = itertools.repeat(None)
+            if isinstance(circuit, QuantumCircuit):
+                circuit = [circuit]
+            if isinstance(observables, (BaseOperator, PauliSumOp, str)):
```

Comment:
Since opflow is deprecated and will be removed soon, we might be able to remove it.
```suggestion
            if isinstance(observables, (BaseOperator, str)):
```